src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create book " + book, e);
+        }
+
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql)) {
+            statement.setLong(1, id);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    return Optional.of(getBook(resultSet));
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find by id for id = " + id, e);
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT * FROM books";
+
+        try (PreparedStatement statement = ConnectionUtil
+                        .getConnection().prepareStatement(sql);
+                ResultSet resultSet = statement.executeQuery()) {
+
+            while (resultSet.next()) {
+                books.add(getBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't retrieve all books", e);
+        }
+
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET price = ?, title = ? WHERE id = ?";
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql)) {
+            statement.setInt(1, book.getPrice());
+            statement.setString(2, book.getTitle());
+            statement.setLong(3, book.getId());
+            if (statement.executeUpdate() > 0) {
+                return book;
+            } else {
+                throw new DataProcessingException("No book found with id = " + book.getId(),
+                        new RuntimeException());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book with id = " + book.getId(), e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM books WHERE id = ?";
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book with id = " + id, e);
+        }
+    }
+
+    private Book getBook(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        Integer price = resultSet.getObject("price", Integer.class);
+        String title = resultSet.getString("title");
+        return new Book(id, title, price);
```

Comment:
According to the checklist, when you convert `ResultSet` to `Book`, it's better to create an object using setters or constructor but not both of them. So, let's use setters or constructor to create a `Book` object.